### ISSUE TITLE
HTTP API docs?

### ISSUE THREAD

**Issue by dessalines** [bookwyrm-social/bookwyrm#2292]
I can't find the location of bookwyrm's API docs anywhere. This would be useful to have, for such things as adding books, getting ratings, etc.

**Comment by mouse-reeve** [comment]
Agreed! The API access is limited right now to the ActivityPub API (so you can't do any authenticated actions, like rating a book, without using the UI), which is why I haven't prioritized documenting it so far, but you're quite right that it's needed.

**Comment by mouse-reeve** [comment]
Unfortunately my comment above is still the case -- we don't have an API (even though it would be great and I'd like to)

**Comment by SMillerDev** [comment]
I've never really touched Django, but I might give it a try for this

**Comment by mouse-reeve** [comment]
> I've never really touched Django, but I might give it a try for this

Thank you, that would be great! If you go for it, I'd recommend biting of a small chunk and opening a PR early, since it's a big undertaking in total. API authentication might be a place to start, but that's just off the top of my head

**Comment by SMillerDev** [comment]
For API Authentication, I'm thinking https://django-oauth-toolkit.readthedocs.io/en/latest/install.html will be a good thing to add. We can keep authenticating applications limited in the first implementation, and extend the application registration endpoint to be public like it is for Mastodon later.

If you have no major gripes with using OAuth and that library I'll start putting it in a PR.
